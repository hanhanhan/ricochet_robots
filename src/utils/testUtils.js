import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom"
import { DndProvider } from "react-dnd"
import { HTML5Backend } from "react-dnd-html5-backend"

import { PlayerContext, usePlayerTurn } from "../components/game"

const AllTheProviders = ({ children }) => {
  const { myTurn, setMyTurn } = usePlayerTurn()
  return (
    <DndProvider backend={HTML5Backend}>
      <PlayerContext.Provider value={{ myTurn, setMyTurn }}>
        {children}
      </PlayerContext.Provider>
    </DndProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
