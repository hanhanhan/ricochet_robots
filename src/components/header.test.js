import React from "react"
import Header from "./header"
import { render } from "../utils/testUtils"

describe("Header", () => {
  it("renders correctly", () => {
    const header = render(<Header siteTitle="Default Starter" />)
    expect(header).toMatchSnapshot()
  })
})
