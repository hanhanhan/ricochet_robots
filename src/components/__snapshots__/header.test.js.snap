// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<body>
  <div>
    <header
      style="background: rebeccapurple; margin-bottom: 0.5rem;"
    >
      <div
        style="margin: 0px auto; max-width: 960px; padding: 0.5rem 1.0875rem;"
      >
        <h1
          style="margin: 0px;"
        >
          <a
            href="/"
            style="color: white; text-decoration: none;"
          >
            Default Starter
          </a>
        </h1>
      </div>
    </header>
  </div>
</body>
`;
